components/net/http_loader.rs

```diff
@@ -412,17 +413,10 @@ fn obtain_response(
     // TODO(#21261) connect_start: set if a persistent connection is *not* used and the last non-redirected
     // fetch passes the timing allow check
     let connect_start = precise_time_ms();
-    // https://url.spec.whatwg.org/#percent-encoded-bytes
+    let url_bytes = url.clone().into_url().as_ref().bytes().collect::<Vec<u8>>();
     let request = HyperRequest::builder()
         .method(method)
-        .uri(
-            url.clone()
-                .into_url()
-                .as_ref()
-                .replace("|", "%7C")
-                .replace("{", "%7B")
-                .replace("}", "%7D"),
-        )
+        .uri(percent_decode(&url_bytes).decode_utf8_lossy().into_owned())
```

Comment:
Why the lossy decoding here? Hyper doesn't need a `str`, maybe use `PercentDecode::if_any`.